bodo/io/iceberg/catalog/dir.py

```diff
@@ -0,0 +1,291 @@
+"""
+A simpler implementation of the Iceberg Java Hadoop Catalog.
+Only supports loading, creating/replacing/appending to tables
+from a directory structure.
+"""
+
+import pyarrow as pa
+from pyiceberg.catalog import (
+    WAREHOUSE_LOCATION,
+    Catalog,
+    MetastoreCatalog,
+    PropertiesUpdateSummary,
+)
+from pyiceberg.exceptions import NoSuchTableError
+from pyiceberg.io import load_file_io
+from pyiceberg.partitioning import UNPARTITIONED_PARTITION_SPEC, PartitionSpec
+from pyiceberg.schema import Schema
+from pyiceberg.serializers import FromInputFile
+from pyiceberg.table import (
+    CommitTableResponse,
+    CreateTableTransaction,
+    StagedTable,
+    Table,
+    sorting,
+)
+from pyiceberg.table.metadata import new_table_metadata
+from pyiceberg.table.update import (
+    TableRequirement,
+    TableUpdate,
+    update_table_metadata,
+)
+from pyiceberg.typedef import EMPTY_DICT, Identifier, Properties
+
+
+class DirCatalog(Catalog):
+    def __init__(self, name: str, **properties: str):
+        super().__init__(name, **properties)
+        if WAREHOUSE_LOCATION not in self.properties:
+            raise ValueError(f"Missing {WAREHOUSE_LOCATION} property")
+
+    @property
+    def warehouse_path(self) -> str:
+        return self.properties[WAREHOUSE_LOCATION]
+
+    def _table_path(self, identifier: Identifier) -> str:
+        wh_path = self.warehouse_path.removesuffix("/")
+        return f"{wh_path}/{'/'.join(identifier)}"
```

Comment:
Bug here to remove cases where `{warehouse_path}//...`

Reply:
To clarify it was a bug in my own code in the base read PR that was discovered when running nightly.